File: nodeJs/src/app/architecture/request.js
class Request {
  constructor(rawData) {
    this.rawData = rawData;
  }

  getList() {
    return this.rawData.split(",")
      .map(s => s.trim())
      .map(s => s.match(/\d/) ? parseInt(s, 10) : s);
  }

  get() {
    return this.rawData;
  }
}

module.exports = Request;

File: java-maven/src/main/java/com/github/softwareeconomics/primaryportskata/Request.java
package com.github.softwareeconomics.primaryportskata;

import java.util.List;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;

public class Request {
  private final String rawData;

  public Request(String rawData) {
    this.rawData = rawData;
  }

  public String get() {
    return rawData;
  }

  public List<Integer> getIntList() {
    return Stream.of(rawData.split(","))
        .map(Integer::parseInt)
        .collect(toList());
  }
}
